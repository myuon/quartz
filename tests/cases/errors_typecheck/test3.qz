fun main() {
    return struct {
        x: 10,
        y: "foo",
        z: true,
    };
}
