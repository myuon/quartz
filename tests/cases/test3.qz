fun calc(b: i32): i32 {
    let a = 1;
    let z = 10;
    return z + a + b;
}

fun main(): i32 {
    return calc(2);
}

