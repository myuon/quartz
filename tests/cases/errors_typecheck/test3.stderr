TypeError unification failed: struct {x: i32, y: string, z: bool} != nil at main[L1:11-L5:5]

0 | fun main() {
1 |     return struct {
1 |            ^^^^^^^^^
2 |         x: 10,
2 | ^^^^^^^^^^^^^^
3 |         y: "foo",
3 | ^^^^^^^^^^^^^^^^^
4 |         z: true,
4 | ^^^^^^^^^^^^^^^^
5 |     };
5 | 
